### PR TITLE
vc re design intermediate

### PR DESCRIPTION
**Pull request by craig8**
Pull request updates 
- remote agent creation workflow and documentation
- initial css for default pages
- peers are shown in both rabbit and zmq systems.
- observer changes are now shown only in warn state

#1911, #1962, #1930

### CODE REVIEW

File: volttron/platform/web/webapp.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def client_closed(self, client, endpoint, identity,
             self.masterweb.vip.rpc.call(identity, 'client.closed', endpoint)
 
     def create_ws_endpoint(self, endpoint, identity):
-        #_log.debug()print(endpoint, identity)
+        # _log.debug()print(endpoint, identity)
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttron/platform/web/webapp.py]: yup

File: volttrontesting/fixtures/volttron_platform_fixtures.py [hunk 2]
```diff
@@ -49,8 +49,9 @@ def get_rand_ipc_vip():
 
 
 def build_wrapper(vip_address, **kwargs):
-    wrapper = PlatformWrapper(message_bus=kwargs.pop('message_bus', None),
-                              ssl_auth=kwargs.pop('ssl_auth', False))
+    wrapper = PlatformWrapper(messagebus=kwargs.pop('message_bus', None),
+                              ssl_auth=kwargs.pop('ssl_auth', False),
+                              instance_name=kwargs.pop('instance_name', 'volttron_test'))
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttrontesting/fixtures/volttron_platform_fixtures.py]: Looks fine

File: volttron/platform/vip/agent/subsystems/auth.py [hunk 3]
```diff
@@ -148,50 +149,77 @@ def connect_remote_platform(self, address, server_key=None,
...
-                            with open("keystore.json", 'w') as fp:
-                                fp.write(json.dumps(KeyStore.generate_keypair_dict()))
-
-                        with open("keystore.json") as fp:
-                            keypair = json.loads(fp.read())
-
-                    value = build_agent(agent_class=agent_class,
-                                        identity=remote_identity,
+                # if the current message bus is zmq then we need
+                # to connect a zmq on the remote, whether that be the
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttron/platform/vip/agent/subsystems/auth.py]: ya what I have makes total sense...we just didn't read the comments inline and look at the flow of the file at this point.

File: services/core/VolttronCentralPlatform/vcplatform/agent.py [hunk 4]
```diff
@@ -247,40 +253,57 @@ def _configure(self, config_name, action, contents):
...
-            _log.error("Couldn't determine server key and address")
-            return
+        vc_serverkey = qry_vc_serverkey
+        if cfg_vc_serverkey is not None:
+            vc_serverkey = cfg_vc_serverkey
 
+        vc_rmq_ca_cert = qry_vc_rmq_ca_cert
+        if cfg_vc_rmq_ca_cert is not None:
+            vc_rmq_ca_cert = cfg_vc_rmq_ca_cert
+#
```
[the remark this answers is not in the document]
Comment (craig8) [reply services/core/VolttronCentralPlatform/vcplatform/agent.py]: yup

File: services/core/VolttronCentralPlatform/vcplatform/agent.py [hunk 5]
```diff
@@ -292,15 +315,16 @@ def _configure(self, config_name, action, contents):
...
+
         self._registration_state = RegistrationStates.NotRegistered
 
-        if message_bus == 'zmq':
-            self._vc_serverkey = s
-            if not self._vc_address or not self._vc_serverkey:
-                _log.error("vc address and serverkey could not be determined. "
-                           "registration is not allowed.")
-                return
+        # if not self._vc_address and not self._vc_serverkey:
```
[the remark this answers is not in the document]
Comment (craig8) [reply services/core/VolttronCentralPlatform/vcplatform/agent.py]: done

File: services/core/VolttronCentralPlatform/vcplatform/agent.py [hunk 6]
```diff
@@ -489,39 +511,54 @@ def _address_type(self, address):
...
         return parsed.scheme
 
     def _establish_connection_to_vc(self):
 
-        if not self._vc_address:
-            raise ValueError("vc_address was not resolved properly.")
-
-        if not self._vc_address.startswith('amqp') and not self._vc_serverkey:
-            raise ValueError("vc_serverkey was not resolved properly.")
+        # if not self._vc_address:
```
[the remark this answers is not in the document]
Comment (craig8) [reply services/core/VolttronCentralPlatform/vcplatform/agent.py]: done

File: services/core/VolttronCentralPlatform/vcplatform/agent.py [hunk 7]
```diff
@@ -489,39 +511,54 @@ def _address_type(self, address):
...
                     address=self._vc_address,
                     serverkey=self._vc_serverkey,
-                    publickey=self.core.publickey,
-                    secretkey=self.core.secretkey,
-                    volttron_central_address=self._vc_address,
-                    volttron_central_instance_name=volttron_central_instance_name,
+                    rmq_ca_cert=self._vc_rmq_ca_cert,
                     agent_class=VCConnection
                 )
+                #
```
[the remark this answers is not in the document]
Comment (craig8) [reply services/core/VolttronCentralPlatform/vcplatform/agent.py]: done

File: services/core/VolttronCentralPlatform/vcplatform/agent.py [hunk 8]
```diff
@@ -537,10 +574,16 @@ def _establish_connection_to_vc(self):
...
-                self._vc_connection.set_main_agent(self)
-                self._still_connected()
-                self._publish_stats()
-                self._publish_device_health()
+                if self._vc_connection is not None:
+                    self._vc_connection.set_main_agent(self)
+                    self._still_connected()
+                    self._publish_stats()
+                    self._publish_device_health()
+                else:
```
[the remark this answers is not in the document]
Comment (craig8) [reply services/core/VolttronCentralPlatform/vcplatform/agent.py]: did this.  now we  have a while loop...

File: volttron/platform/web/admin_endpoints.py [hunk 9]
```diff
@@ -127,21 +131,50 @@ def __api_endpoint(self, endpoint, data):
...
             data = dict(status=self._certs.get_csr_status(common_name),
                         cert=self._certs.get_cert_from_csr(common_name))
         except ValueError as e:
             data = dict(status="ERROR", message=e.message)
 
         return Response(json.dumps(data), content_type="application/json")
 
+    def __deny_csr_api(self, common_name):
+        try:
+            self._certs.deny_csr(common_name)
```
Comment (schandrika) [inline volttron/platform/web/admin_endpoints.py]: Remove rabbitmq user on deny

**Pull request merged**